### PR TITLE
Merge Sidepanel into Modal

### PR DESCRIPTION
**Pull request by GuillaumeNury**
## Description

Two implementations only differ on css classes:
* add `modal.open(component, data,  { mode: 'sidepanel' })`
* deprecate `LuSidepanel`

-----

Can be tested in [SidepanelStories](http://lucca-front.lucca.local/PR-1918/storybook/?path=/story/documentation-overlays-sidepanel--basic).

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1918/storybook

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1918/storybook

File: packages/ng/modal/src/lib/modal-panel.component.ts [hunk 1]
```diff
@@ -38,8 +42,12 @@ export abstract class ALuModalPanelComponent<T extends ILuModalContent> implemen
 	public readonly modalId = modalId++;
 
 	private _subs = new Subscription();
+	public modalClasses: LuModalClasses;
 
-	constructor(protected _ref: ALuModalRef<T>, protected _cdr: ChangeDetectorRef) {}
+	constructor(protected _ref: ALuModalRef<T>, protected _cdr: ChangeDetectorRef, _elementRef: ElementRef<HTMLElement>, _renderer: Renderer2) {
```
Comment (shajz) [inline packages/ng/modal/src/lib/modal-panel.component.ts:47]:
```suggestion
	constructor(protected _ref: ALuModalRef<T>, _elementRef: ElementRef<HTMLElement>, _renderer: Renderer2) {
```

`ChangeDetectorRef` isn't needed anymore (yay!)

**Review by shajz — APPROVED** [review]

**Review by AlexiaAlarza — DISMISSED** [review]

File: packages/ng/modal/src/lib/modal-panel.component.ts (hunk 1 above)
Comment (GuillaumeNury) [reply packages/ng/modal/src/lib/modal-panel.component.ts]: Oh thx!

**Review by AlexiaAlarza — APPROVED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1918/storybook

**Review by shajz — APPROVED** [review]

**Pull request merged**